roles/ansible-test/tasks/init_collection.yaml

```diff
@@ -4,8 +4,36 @@
     _test_location: "{{ ansible_test_collection_dir }}/{{ ansible_test_collection_namespace }}/{{ ansible_test_collection_name }}"
     _test_cfg_location: "{{ ansible_test_collection_dir }}/{{ ansible_test_collection_namespace }}/{{ ansible_test_collection_name }}/tests/integration/{{ ansible_test_test_command }}.cfg"
 
+- name: Setup minimum test requirements
+  set_fact:
+    _test_requirements: '-r {{ _test_location }}/requirements.txt -r {{ _test_location }}/test-requirements.txt'
+
+- name: Check for existence of test command specific requirements/constraints
+  when: ansible_test_test_command in _file_locations
+  vars:
+    _file_locations:
+      integration: '{{ _test_location }}/tests/integration'
+      units: '{{ _test_location }}/tests/unit'
```

Comment:
Can we standardize on one of these two files?

Reply:
We can, I'd gone for splitting it like this to be consistent with the behaviour of the containerized ansible test

Reply:
Not really as long as we use ansible-test to execute the tests.  These locations are picked up by ansible-test in and used to install the needed requirements for the given type of test.

Reply:
Ideally, the way I see this working is we create github.com/ansible-collections/requirements repo. In it, we store an upper-constraints.txt file, along with other lower constraints and caps.

unit / sanity / integration tests should use the same constraints, along with all the collections we ship in an execution environment.  The goal, is to not only ensure each collection works with the specific libraries but all of our collections are installable together with the same constraints.

https://docs.openstack.org/project-team-guide/dependency-management.html really explains what I am hoping we get too.

Reply:
The use case I was aiming for with the test constraints was to ensure that the AWS modules work with the minimum versions of boto3/botocore we claim to support.  Amazon add new features very rapidly, and when we don't explicitly check it's easy to accidentally rely on a newer feature and break things for people trying to use the minimum version we say we support.

As such the test constraint files are going to be specific to the version of the collection, this doesn't work particularly well for storing the test constraints outside of the collection repo.